.github/workflows/validate-pr-actions.yaml

```diff
@@ -2,34 +2,109 @@
-      image:
-        type: string
-        required: false
   pull_request:
     paths:
+      - Dockerfile
+      - src/**
+      - Pipfile
+      - Pipfile.lock
       - actions/**
       - mpyl_config.example.yml
       - tests/projects/job/**
       - tests/test_resources/changed-files/*
 
 # If multiple commits are pushed simultaneously, only build the last one.
 concurrency:
-  group: ${{ github.workflow }}-${{ github.event.pull_request.number || inputs.image }}
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number }}
   cancel-in-progress: false
 
 jobs:
+  changed-files:
+    name: 📋 Do we need to build?
+    runs-on: ubuntu-latest
+    outputs:
+      build-docker-image: ${{ steps.changed-files.outputs.any_changed == 'true' }}
+    steps:
+      - uses: tj-actions/changed-files@v45.0.4
+        id: changed-files
+        with:
+          files: |-
+            Dockerfile
+            src/**
+            Pipfile
+            Pipfile.lock
+
+  docker:
+    name: 🐳 Build & publish Docker image
+    runs-on: ubuntu-latest
+    needs: changed-files
+    if: needs.changed-files.outputs.build-docker-image == 'true'
+    outputs:
+      image: ${{ steps.docker-image.outputs.image }}
+    steps:
+      - uses: actions/checkout@v4.2.2
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4.0.2
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ECR_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_ECR_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+
+      - name: Login to Amazon ECR Public
+        id: login-ecr-public
+        uses: aws-actions/amazon-ecr-login@v2.0.1
+        with:
+          registry-type: public
+
+      - name: Create repository in ECR
+        uses: int128/create-ecr-repository-action@v1.329.0
+        with:
+          repository: gh-mpyl
+          public: true
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3.2.0
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3.7.1
+
+      - name: Compute Docker image
+        run: echo "image=$REGISTRY/$REPOSITORY/$PROJECT:$VERSION" | tee -a "$GITHUB_OUTPUT"
+        id: docker-image
+        env:
+          REGISTRY: ${{ steps.login-ecr-public.outputs.registry }}
+          REPOSITORY: ${{ vars.AWS_ECR_PUBLIC_ALIAS }}
+          PROJECT: gh-mpyl
+          VERSION: pr-${{ github.event.pull_request.number }}
+
+      - name: Build and push
+        id: docker-build-push
+        uses: docker/build-push-action@v6.9.0
+        with:
+          file: Dockerfile
+          context: .
+          tags: ${{ steps.docker-image.outputs.image }}
+          platforms: |-
+            linux/amd64
+            linux/arm64
+          provenance: false
+          push: true
+
   health-check:
     name: ▶️ Run actions/health-check
     runs-on: ubuntu-latest
+    if: ${{ !failure() && !cancelled() }}
+    needs:
+      - changed-files
+      - docker
```

Comment:
So how does this work when `docker` is skipped?

Reply:
That's why we have the `if` on the line before:
```yaml
if: ${{ !failure() && !cancelled() }}
needs:
  - docker
```

Meaning: run after the `docker` job, and only if it didn't fail or was cancelled (i.e.  if it completed successfully or if it was skipped).